common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java

```diff
@@ -307,7 +307,18 @@ public static boolean isValidBrokerRedirect(@Nullable final String redirectUri,
                                                 @NonNull final String packageName) {
         final String methodName = ":isValidBrokerRedirect";
         final String expectedBrokerRedirectUri = getBrokerRedirectUri(context, packageName);
-        final boolean isValidBrokerRedirect = StringUtil.equalsIgnoreCase(redirectUri, expectedBrokerRedirectUri);
+        boolean isValidBrokerRedirect = StringUtil.equalsIgnoreCase(redirectUri, expectedBrokerRedirectUri);
+        if (packageName.equals(AuthenticationConstants.Broker.AZURE_AUTHENTICATOR_APP_PACKAGE_NAME))
+        {
+            final PackageHelper info = new PackageHelper(context.getPackageManager());
+            final String signatureDigest = info.getCurrentSignatureForPackage(packageName);
+            if(signatureDigest.equals(BrokerData.MICROSOFT_AUTHENTICATOR_PROD.signatureHash)
+                || signatureDigest.equals(BrokerData.MICROSOFT_AUTHENTICATOR_DEBUG.signatureHash)) {
```

Comment:
probably want to flip this check to avoid potential NPE

Reply:
Or use the utility methods that do null safe comparisons.